spyder/plugins/onlinehelp/widgets.py

```diff
@@ -46,155 +71,421 @@ def spyder_safeimport(path, forceload=0, cache={}):
+        self.start_server()
+
+    def is_server_running(self):
+        """Return True if pydoc server is already running."""
+        return self.server is not None
+
     def start_server(self):
-        """Start pydoc server"""
+        """Start pydoc server."""
         if self.server is None:
-            self.port = select_port(default_port=self.DEFAULT_PORT)
-            self.set_home_url('http://127.0.0.1:%d/' % self.port)
-        elif self.server.is_running():
-            self.quit_server()
-        self.server = PydocServer(port=self.port)
-        self.server.server_started.connect(self.initialize_continued)
+            self.set_home_url('http://127.0.0.1:{}/'.format(PORT))
+        elif self.server.isRunning():
+            self.server.sig_server_started.disconnect(
+                self._continue_initialization)
+            self.server.quit()
+
+        self.server = PydocServer(port=PORT)
+        self.server.sig_server_started.connect(
+            self._continue_initialization)
         self.server.start()
 
     def quit_server(self):
-        """Quit the server"""
+        """Quit the server."""
         if self.server is None:
             return
+
         if self.server.is_running():
-            self.server.server_started.disconnect(self.initialize_continued)
+            self.server.sig_server_started.disconnect(
+                self._continue_initialization)
             self.server.quit_server()
             self.server.quit()
 
-    # ------ WebBrowser API ---------------------------------------------------
     def get_label(self):
         """Return address label text"""
         return _("Package:")
 
     def reload(self):
-        """Reload page"""
+        """Reload page."""
         self.start_server()
         self.webview.reload()
 
-    def load_finished(self, ok):
-        """Handle load finished."""
-        pass
-
     def text_to_url(self, text):
-        """Convert text address into QUrl object"""
+        """
+        Convert text address into QUrl object.
+
+        Parameters
+        ----------
+        text: str
+            Url address.
+        """
         if text != 'about:blank':
             text += '.html'
+
         if text.startswith('/'):
             text = text[1:]
 
         return QUrl(self.home_url.toString() + text)
 
     def url_to_text(self, url):
-        """Convert QUrl object to displayed text in combo box"""
+        """
+        Convert QUrl object to displayed text in combo box.
+
+        Parameters
+        ----------
+        url: QUrl
+            Url address.
+        """
         string_url = url.toString()
         if 'about:blank' in string_url:
             return 'about:blank'
         elif 'get?key=' in string_url or 'search?key=' in string_url:
-            return ''
-        return osp.splitext(to_text_string(url.path()))[0][1:]
+            return url.toString().split('=')[-1]
+
+        return os.path.splitext(str(url.path()))[0][1:]

```

Comment:
Please use `osp` instead of `os.path` here.

Reply:
Fixed